scripts/macos/agent_diagnostics.sh

```diff
@@ -25,47 +26,62 @@ if [[ ! -d "${JCPATH}" ]]; then
 fi
 
 # Is zip installed?
-if ! which zip 2> /dev/null; then
+if ! which zip &> /dev/null; then
   ZPATH="false"
 else
   ZPATH=$(which zip)
 fi
 
 function indent() {
-	sed 's/^/		/g'
+	sed 's/^/'$'\t'$'\t/g'
 }
 
 function zipjc() {
   # Take inventory of files to be zipped.
   for i in "${JCPATH}"/*; do
-    INVENTORY+=("${i}")
+    if [[ "${i}" != *.crt* ]] && [[ "${i}" != *.key* ]]; then
+      INVENTORY+=("${i}")
+    fi
   done
 
   # check to see if zip exists.
-  if [[ "$ZPATH" = "false" ]]; then
-    ZIPIT="zip is not installed. please send the following files with your support request:\n${INVENTORY[*]}"
+  if [[ "${ZPATH}" = "false" ]]; then
+    if [[ -f "${TARFILE}" ]]; then
+      mv "${TARFILE}" ./jc"${STAMP}".bak.tar
+      tar -rf "${TARFILE}" "${INVENTORY[@]}" 1> /dev/null
+    else
+      ZIPIT="${TARFILE} has been created containing the following files:"
+      tar -rf "${TARFILE}" "${INVENTORY[@]}" 1> /dev/null
+    fi
   else
     if [[ -f "${ZIPFILE}" ]]; then
       mv "${ZIPFILE}" ./jc"${STAMP}".bak.zip
-      zip -r "*.crt" -r "${ZIPFILE}" "${JCPATH}" > /dev/null 1
+      zip -r "${ZIPFILE}" "${INVENTORY[@]}" 1> /dev/null
     else
-      ZIPIT="${ZIPFILE} has been created, containing the following files:"
-      zip -x "*.crt" -r "${ZIPFILE}" "${JCPATH}" > /dev/null 1
+      ZIPIT="${ZIPFILE} has been created containing the following files:"
+      zip -r "${ZIPFILE}" "${INVENTORY[@]}" 1> /dev/null
     fi
   fi
 }
 
 function ziplog() {
   # Zip the log files. 
-  LOGFILES=("jcagent.log" "jcUpdate.log" "jclocalclient.log" "jctray.log" "jumpcloud-loginwindow/*")
-  for i in "${LOGFILES[@]}"; do
-    if [ -f "${JCLOG}""${i}" ]; then
-      zip "${ZIPFILE}" "${JCLOG}""${i}" > /dev/null 1
-      LOGIT+=("${JCLOG}${i} has been successfully added to ${ZIPFILE}.")
-    else
-      LOGIT+=("${JCLOG}${i} doesn't exist.")
-    fi
-  done
+  LOGFILES=("jcagent.log" "jcUpdate.log" "jclocalclient.log" "jctray.log" "jumpcloud-loginwindow")
+  if [[ "${ZPATH}" = "false" ]]; then
+    for i in "${LOGFILES[@]}"; do
+      if [[ -f "${JCLOG}""${i}" ]] || [[ -d "${JCLOG}""${i}" ]]; then
+        tar -rf "${TARFILE}" "${JCLOG}""${i}" 1> /dev/null
+        LOGIT+=("${JCLOG}${i} has been added to ${TARFILE}.")
+      fi
+    done
+  else
+    for i in "${LOGFILES[@]}"; do
+      if [[ -f "${JCLOG}""${i}" ]] || [[ -d "${JCLOG}""${i}" ]]; then
+        zip -r "${ZIPFILE}" "${JCLOG}""${i}" > /dev/null 1
```

Comment:
The redirection here should be consistent with the redirection on the line above, `1> /dev/null`.

Reply:
Yes it should, and now it is.